### ISSUE TITLE
Different results from FoVOrthographicCameras between 0.4.0 and 0.5.0/0.6.0

### ISSUE THREAD

**Issue by pomelyu** [facebookresearch/pytorch3d#965]
## 🐛 Bugs / Unexpected behaviors
My problem is similar to #838 and #842, but I use `FovOrhographicCameras` instead of `FovPerspectiveCameras`.  The image is squeezed after updating to version above 0.4.0.
<img width="816" alt="image" src="https://user-images.githubusercontent.com/2679561/144207984-27db8348-7eed-4f74-b8d1-92aa4f7a3570.png">
I can get the original result by adding the `scale_xyz=((IMAGE_RATIO, 1, 1),)` to the camera initialization. However, it looks like a workaround. I tried to explain the code as follow: `xmin, xmax, ymin, ymax` constructs the view frustum. If width != height, each square pixel in the images will stand for the rectangle region in the frustum. As a result, I need to rescale the rectangle region to be the square. Is it correct or is it expected?

By the way, I am confused about the concept behind `FovOrhographicCameras` and `FovPerspectiveCameras` after 0.4.0. It seems the `aspect_ratio` in `FovPerspectiveCameras` and `scale_xyz` in `FovOrhographicCameras` are both related to the pixels(pixel_aspect_ratio?) instead of the camera view or the film size. Could you help me to clarify these terms, thanks.

## Instructions To Reproduce the Issue:
The code is almost the same as what described in #838, except the camera
```python
import os
import numpy
import torch
import cv2
import numpy as np

# USER PARAMETERS
IMAGE_HEIGHT = 300
IMAGE_WIDTH = 512
IMAGE_RATIO = IMAGE_WIDTH / IMAGE_HEIGHT

# Data structures and functions for rendering
from pytorch3d.io import load_objs_as_meshes, load_obj
from pytorch3d.renderer import (
    BlendParams,
    look_at_view_transform,
    FoVPerspectiveCameras,
    FoVOrthographicCameras,
    PointLights,
    RasterizationSettings,
    MeshRenderer,
    MeshRasterizer,
    SoftPhongShader,
)
import urllib.request

# Download cow
urls = ['https://dl.fbaipublicfiles.com/pytorch3d/data/cow_mesh/cow.obj',
        'https://dl.fbaipublicfiles.com/pytorch3d/data/cow_mesh/cow.mtl',
        'https://dl.fbaipublicfiles.com/pytorch3d/data/cow_mesh/cow_texture.png']

names = ['cow.obj', 'cow.mtl', 'cow_texture.png']
for i in range(3):
    if os.path.exists(names[i]) == False:
        urllib.request.urlretrieve(urls[i], names[i])

# Create device
device = torch.device("cuda:0")

# Load mesh
mesh = load_objs_as_meshes(["cow.obj"], device=device)

# Initialize camera.
R, T = look_at_view_transform(2.7, 0, 180)
cameras = FoVOrthographicCameras(device=device, R=R, T=T, min_x=-IMAGE_RATIO, max_x=+IMAGE_RATIO)
# fix the problem after 0.4.0
# cameras = FoVOrthographicCameras(device=device, R=R, T=T, min_x=-IMAGE_RATIO, max_x=+IMAGE_RATIO, scale_xyz=((IMAGE_RATIO, 1, 1),))

# Initialize render
raster_settings = RasterizationSettings(
    image_size=(IMAGE_HEIGHT, IMAGE_WIDTH),
    blur_radius=0.0,
    faces_per_pixel=1,
)
lights = PointLights(device=device, location=[[0.0, 0.0, -3.0]])
rasterizer=MeshRasterizer(
    cameras=cameras,
    raster_settings=raster_settings
)
blend_params = BlendParams(sigma=1e-4, gamma=1e-4, background_color=(0.0, 0.0, 0.0))
shader = SoftPhongShader(
        device=device,
        cameras=cameras,
        blend_params=blend_params,
        lights=lights)

renderer = MeshRenderer(
    rasterizer, shader)

# Render image
images = renderer(mesh)
image = images[0, ..., :3].cpu().numpy()

cv2.imwrite("cow.jpg", (image[..., ::-1] * 255).astype(np.uint8))
```

**Comment by gkioxari** [comment]
Here are a few tips when using renderers and cameras in PyTorch3D (and in other libraries). They detail how one should go about thinking renderings and cameras in the context of the libraries used and its conventions.

- In PyTorch3D, non-square rasterization rasterizers points in normalized space (NDC) such that the smallest side ranges in `[-1, +1]` and the largest side is in `[-s, s]` where `s=ASPECT_RATIO > 1`. For square images, that space is always `[-1, +1]x[-1, +1]`. Different libraries have different conventions for non-square images; this is the convention in PyTorch3D.

- Cameras transform points to NDC space. In your case, when you define `min_x` and `max_x` with the `IMAGE_RATIO` you are effectively scaling the object in one side (shrinking it in the x-axis as `IMAGE_RATIO > 1`) which the camera maps `[-1, 1]`. Our renderer takes this and renders it and the cow is squished in the x-axis per your result. That is correct. 

- If you instead don't scale the object and define `cameras = FoVOrthographicCameras(device=device, R=R, T=T, min_x=-1.0, max_x=+1.0)` then you get the image below
![image2](https://user-images.githubusercontent.com/4369065/144335623-cbfa23d9-98a0-4a21-b470-734824983d1d.png)

Note that this is the same as your output when you apply the `scale_xyz` which undoes the scaling of the cow in the x-axis.

**Comment by bottler** [comment]
(Note that cameras for non square images before the version 0.5.0 transformed points to NDC [-1,1]x[-1,1]  ).

**Comment by pomelyu** [comment]
Thanks for the kindly reply.

Finally I figure out correctly format. Tell me if I am wrong.
For the perspective camera. The following statements produce same results
```python
IMAGE_HEIGHT = 300
IMAGE_WIDTH = 512

# since IMAGE_HEIGHT < IMAGE_WIDTH
# in ndc space IMAGE_HEIGHT, IMAGE_WIDTH will be mapped to [-1, 1] and [-u, u] respectively

# cow mesh is located between [-1, 1] in all the three dimension
mesh = load_objs_as_meshes(["cow.obj"], device=device)
R, T = look_at_view_transform(eye=((0, 0, -2.7),))
fov = 60

# FOVPerspectiveCameras(in NDC space)
cameras = FoVPerspectiveCameras(device=device, R=R, T=T, aspect_ratio=1, fov= fov)

# PerspectiveCameras in world space
focal_length = IMAGE_HEIGHT * 0.5 / np.tan(np.radians(fov*0.5))
cameras = PerspectiveCameras(device=device, R=R, T=T, in_ndc=False, focal_length=focal_length, image_size=((IMAGE_HEIGHT, IMAGE_WIDTH),), principal_point=((IMAGE_WIDTH/2, IMAGE_HEIGHT/2),))

# PerspectiveCameras in NDC space
focal_length = 2 * 0.5 / np.tan(np.radians(fov*0.5))
cameras = PerspectiveCameras(device=device, R=R, T=T, in_ndc=True, focal_length=focal_length, image_size=((IMAGE_HEIGHT, IMAGE_WIDTH),))

# PerspectiveCameras in world space(after enlarge)
# as the object enlarge twice, we should increase the camera distance twice
mesh = mesh.scale_verts(2.0)
R, T = look_at_view_transform(eye=((0, 0, -2.7*2),))
focal_length = IMAGE_HEIGHT * 0.5 / np.tan(np.radians(fov*0.5))
cameras = PerspectiveCameras(device=device, R=R, T=T, in_ndc=False, focal_length=focal_length, image_size=((IMAGE_HEIGHT, IMAGE_WIDTH),), principal_point=((IMAGE_WIDTH/2, IMAGE_HEIGHT/2),))
```

And for the orthographic camera. The following statements produce same results
```python
# FoVOrthographicCameras(in NDC space)
cameras = FoVOrthographicCameras(device=device, R=R, T=T)

# OrthographicCameras in world space
cameras = OrthographicCameras(device=device, R=R, T=T, in_ndc=False, focal_length=IMAGE_HEIGHT/2, image_size=((IMAGE_HEIGHT, IMAGE_WIDTH),), principal_point=((IMAGE_WIDTH/2, IMAGE_HEIGHT/2),))

# OrthographicCameras in NDC space
cameras = OrthographicCameras(device=device, R=R, T=T, in_ndc=True, focal_length=1, image_size=((IMAGE_HEIGHT, IMAGE_WIDTH),))

# OrthographicCameras in world space(after enlarge)
mesh = mesh.scale_verts(2.0)
cameras = OrthographicCameras(device=device, R=R, T=T, in_ndc=False, focal_length=(IMAGE_HEIGHT/2)/2, image_size=((IMAGE_HEIGHT, IMAGE_WIDTH),), principal_point=((IMAGE_WIDTH/2, IMAGE_HEIGHT/2),))
```

The first problem is how to realize `focal_length` in `OrthographicCameras`. I think the term should be modified to `scale`, since It is the ration between the camera view and the size of NDC[-1, 1].

The other problem is the term **NDC space**. [To my knowledge](https://learnopengl.com/Getting-started/Coordinate-Systems), object is transformed to NDC space through the world-to-camera matrix and camera projection matrix. Hence, there is no projection matrix for camera in NDC coordinate. In this coordinate, there is also no different between perspective and orthogonal camera, since they have already been projected. For the camera in NDC space, we only need view transform matrix to generate the final rendering result. I think the term **NDC space** in pytorch3d simply means the normalized global coordinates not the ndc coordinates after projection. That's what confused me.
